pkg/kubectl/cluster.go

```diff
@@ -30,6 +31,8 @@ import (
 type ClusterGeneratorV1Beta1 struct {
 	// Name of the cluster context (required)
 	Name string
+	// Name of the federation this cluster is joining (required)
```

Comment:
Just to close this discussion, summarizing our offline discussion here:

> To clarify, the code here is kubectl and not kubefed. We have defined policies for kubectl version skew support that we cant break.

`ClusterGenerator` is not kubectl code, it is kubefed code in the sense that kubefed is the only client. It sits in kubectl package to reuse some of the machinery there.

> FWIW, we do need to be strict about not breaking kubefed as well (now that its beta)

Agreed. But this doesn't break kubefed too. It is just changing the internal detail.

> kubefed should not be using ClusterGenerator. Its providing the yaml I believe.

kubefed is using ClusterGenerator. In fact, it is the only consumer of that interface. It uses the kubectl generator machinery to simplify a number of other things (read: reinvent the wheel).

But we decided to remove the annotation from cluster object because it is unnecessary code.